Agregar endpoints para restauración de datos y obtención del usuario autenticado

JunhoRS opened the pull request:
Cambios Incluidos:
Nuevo Endpoint: POST /api/categorias/restore
Restaura categorías, ingresos y gastos marcados como eliminados para el usuario autenticado.
Utiliza el userID del JWT para identificar y restaurar los registros propios del usuario.

Nuevo Endpoint: GET /api/categorias/user/me
Permite obtener la información del usuario autenticado, incluyendo su UsuarioId, Nombre, CorreoElectronico y Rol.
Verifica el userID en el JWT y responde con los datos básicos del perfil del usuario.

JunhoRS committed:
Merge branch 'dev' into feature/usuario-endpoints

JunhoRS committed:
Agregar endpoints de usuario en CategoriasController

SmillRivera14 reviewed (commented):
Se removió el entpoint de GetAuthenticatedUser (Rama features/Refactorizacion):
Debido a: Estaba dentro del controlador de categorías.
Se hizo: Coloque la lógica dentro del GetUsuarios (entpoint que ya traía al usuario, pero, aunque no estuviera autenticado).

JunhoRS committed:
Update CategoriasController.cs

/Categoria/Restaurar/{id}
/Gastos/Restaurar/{id}
/Ingresos/Restaurar/{id}

Bug: los usuarios pueden eliminar ingresos o gastos que no son suyos
Explicación
Solo se pide el ID de estos para eliminar los datos, no importa que sean o no sean del usuario

SmillRivera14 reviewed (commented):
Utilice tu lógica, pero cree un método a parte para poder reutilizar el código.
